### PR TITLE
Dashboards distribution tools

### PR DESCRIPTION
**Pull request by bre1470**
`npx gulp dashboards/dist --dryrun --release 1.0.0` should create:
- [x] build/api/dashboards
- [x] build/dist/dashboards/code 
- [x] build/dist/dashboards/code/css 
- [x] build/dist/dashboards/code/gfx
- [x] build/dist/dashboards/examples
- [x] build/dist/dashboards/js-gzip
- [x] build/dist/dashboards/Dashboards.zip
- [x] changelog/dashboards (https://github.com/highcharts/highcharts/pull/19229)
- [x] code/dashboards
- [x] code/dashboards/dashboards.js
- [x] ../dashboards-dist/ (https://github.com/highcharts/dashboards-dist)
- [x] ../dashboards-dist/README.md
- [x] NPM/@highcharts/dashboards
- [x] S3/dashboards
- [x] Fix global declarations

Reset branch builds with `rm -rf build code cypress js out`.

### CODE REVIEW

**Review by pawelfus — COMMENTED** [review]
Looks good, works good until:

```
[10:54:19] Finished 'dashboards/dist-release' after 5.38 s
[10:54:19] Starting 'dashboards/dist-upload'...
[10:54:19] Not implemented
[10:54:19] Finished 'dashboards/dist-upload' after 1.08 ms
[10:54:19] Finished 'dashboards/dist' after 29 s
```

But I guess that's whats hidden under `S3/dashboards` checkbox in the PR description ;) 

PS: Great job!

File: tools/gulptasks/dashboards/dist-release.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    await processLib.exec(`git commit -m "v${release}"`, {
+        cwd: distRepository
+    });
+    await processLib.exec(`git tag v${release}`, {
+        cwd: distRepository
+    });
+
+    // Do magic
+
+    if (argv.dry) {
```
Comment (goransle) [inline tools/gulptasks/dashboards/dist-release.js:104]: Could we use `dryrun` here to be consistent with other tasks (and the S3 CLI)?

**Comment by pawelfus** [comment]
Tested with:
> npx gulp dashboards/dist --release 1.0.0 --dry --verbose

**bzzt**
> [13:57:21] Starting 'distZip'...
[13:57:21] 'distZip' errored after 2.74 ms
[13:57:21] Error: No valid `--release x.x.x` provided.
    at distZip (/Users/pawelfus/Projects/highcharts/tools/gulptasks/dashboards/dist-zip.js:35:15)
    at bound (node:domain:433:15)
    at runBound (node:domain:444:12)
    at asyncRunner (/Users/pawelfus/Projects/highcharts/node_modules/async-done/index.js:55:18)
    at processTicksAndRejections (node:internal/process/task_queues:78:11)
[13:57:21] 'dashboards/dist-zip' errored after 682 ms
[13:57:21] 'dashboards/dist' errored after 23 s

**Review by pawellysy — APPROVED** [review]
LGTM!

**Review by goransle — APPROVED** [review]

**Review by pawelfus — APPROVED** [review]

**Pull request merged**